### PR TITLE
[fix]: [CDS-52353]: Check if pipelineBranchName is empty in trigger for remote pipeline

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)

### CODE REVIEW

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java` line 101 at r1](https://reviewable.io/reviews/harness/harness-core/43906#-NOByu4s8hciUhWU-kE-:-NOByu4s8hciUhWU-kE0:b-rmk7uq) ([raw file](https://github.com/harness/harness-core/blob/e8dbc324ad9b3c837c741d6e4fe758bf7ff00486/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java#L101)):*
> ```Java
>   }
> 
>   public PMSPipelineResponseDTO fetchPipelineResponseForTrigger(TriggerDetails triggerDetails) {
> ```

Can we make `fetchPipelineForTrigger` call this method, to avoid code repetition? I think it would be nice to rename `fetchPipelineForTrigger ` to `fetchPipelineYamlForTrigger` and rename this method to `fetchPipelineForTrigger`.

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 780 at r1](https://reviewable.io/reviews/harness/harness-core/43906#-NOBzVolEND3TPTcyNB9:-NOBzVolEND3TPTcyNBA:biqlxsn) ([raw file](https://github.com/harness/harness-core/blob/e8dbc324ad9b3c837c741d6e4fe758bf7ff00486/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L780)):*
> ```Java
>     Map<String, Map<String, String>> errorMap = new HashMap<>();
>     if (pipelineResponse != null) {
>       if (pipelineResponse.getGitDetails() != null) {
> ```

Did you confirm if inline pipelines always have null `gitDetails`?

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java [hunk 1]
```diff
@@ -98,6 +98,14 @@ public Optional<String> fetchPipelineForTrigger(TriggerDetails triggerDetails) {
     return response != null ? Optional.of(response.getYamlPipeline()) : Optional.empty();
   }
 
+  public PMSPipelineResponseDTO fetchPipelineResponseForTrigger(TriggerDetails triggerDetails) {
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java:101]: @rathodmeetsatish Do we need this new method?
Isn't this doing the same? `fetchPipelineForTrigger`

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java [hunk 2]
```diff
@@ -98,6 +98,14 @@ public Optional<String> fetchPipelineForTrigger(TriggerDetails triggerDetails) {
...
+    NGTriggerEntity ngTriggerEntity = triggerDetails.getNgTriggerEntity();
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java:102]: @rathodmeetsatish Do we need this new method?
Isn't this doing the same? fetchPipelineForTrigger

**Review by sribalij — CHANGES_REQUESTED** [review]
Added comments

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 6 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java` line 101 at r1](https://reviewable.io/reviews/harness/harness-core/43906#-NOByu4s8hciUhWU-kE-:-NOEy0pIDlPS-sMrGdqs:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e8dbc324ad9b3c837c741d6e4fe758bf7ff00486/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java#L101)):*
<details><summary><i>Previously, vcalasansh (Vinícius Calasans) wrote…</i></summary><blockquote>

Can we make `fetchPipelineForTrigger` call this method, to avoid code repetition? I think it would be nice to rename `fetchPipelineForTrigger ` to `fetchPipelineYamlForTrigger` and rename this method to `fetchPipelineForTrigger`.
</blockquote></details>

Done.
___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java` line 102 at r1](https://reviewable.io/reviews/harness/harness-core/43906#-NOBVacepyYzZyRrkt4n-r1-102:-NOElsKJBfgiG-_nCGpU:b-sq4npp) ([raw file](https://github.com/harness/harness-core/blob/e8dbc324ad9b3c837c741d6e4fe758bf7ff00486/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/buildtriggers/helpers/BuildTriggerHelper.java#L102)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

@rathodmeetsatish Do we need this new method?
Isn't this doing the same? fetchPipelineForTrigger
</blockquote></details>

no it is fetching the yaml

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 8 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 780 at r1](https://reviewable.io/reviews/harness/harness-core/43906#-NOBzVolEND3TPTcyNB9:-NOGlv9u9sna2BiTuE8u:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e8dbc324ad9b3c837c741d6e4fe758bf7ff00486/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L780)):*
<details><summary><i>Previously, vcalasansh (Vinícius Calasans) wrote…</i></summary><blockquote>

Did you confirm if inline pipelines always have null `gitDetails`?
</blockquote></details>

Done.

**Review by sribalij — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 8 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by vcalasansh — APPROVED** [review]
> Reviewed 3 of 6 files at r2, 3 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by vcalasansh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> Reviewed 3 of 6 files at r2, 3 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: :shipit: complete! ✓

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 3 of 6 files at r2, 3 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43906)* status: :shipit: complete! ✓

**Pull request merged**